tests/sgx_zerobase/enc/enc.cpp

```diff
@@ -3,42 +3,76 @@
 
 #include <openenclave/enclave.h>
 #include <openenclave/internal/tests.h>
+#include <iostream>
+
 #include "sgx_zerobase_t.h"
 
-const char* protected_message = "Hello world from Enclave\n\0";
+static uint64_t faulting_address = sizeof(uint64_t) - 1;
+static uint32_t error_code;
+static uint32_t exception_code;
 
-int secure_string_patching(
-    const char* source,
-    char* output,
-    size_t output_length)
+uint64_t test_pfgp_handler(oe_exception_record_t* exception_record)
 {
-    const char* running_source = source;
-    size_t running_length = output_length;
-    while (running_length > 0 && *running_source != '\0')
+    if (exception_record->code == OE_EXCEPTION_PAGE_FAULT)
     {
-        *output = *running_source;
-        running_length--;
-        running_source++;
-        output++;
+        const int MOV_INSTRUCTION_BYTES = 3;
+        faulting_address = exception_record->faulting_address;
+        error_code = exception_record->error_code;
```

Comment:
`error_code` seems not needed and can be removed. But I'm curious what's the value of the error_code, can you show me?

Reply:
The exception error codes are followed as per definition in include/openenclave/bits/exception.h.

1: Test 2: Create 0-base enclave
1: Expected result : OE_OK
1: 
1: Test 3: Test ecall, ocall on 0-base enclave
1: Expected result : OE_OK
1: [host] Message from enclave : testing ocall
1: 
1: Test 4: Check host access between /proc/sys/vm/mmap_min_addr and enclave image start address - [0x1000, 0x21000)
1: Expected result : success
1: 
1: Test 5: Enclave memory access at 0x0
1: Expected result: page fault
1: address 0x0, exception 5
1: 
1: Test 6: memory access between enclave image start address and the end of elrange - [0x21000, 0x40000)
1: Expected result: no exceptions
1: address 0x2f000, exception 0
1: 
1: === passed all tests (sgx_zerobase)
1/1 Test #1: tests/sgx_zerobase ...............   Passed    0.12 sec

Reply:
Removed error_code and made argument 'exception' in ecall a bool.